bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/SelectDataWizardPage.java

```diff
@@ -0,0 +1,389 @@
+        this.selectedDataObservable = selectedDataObservable;
+        this.businessObjectStore = businessObjectStore;
+        this.rootNameObservable = rootNameObservable;
+        this.contract = contract;
+        this.dataTypeSelectionOptions = dataTypeSelectionOptions;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.eclipse.jface.dialogs.IDialogPage#createControl(org.eclipse.swt.widgets.Composite)
+     */
+    @Override
+    public void createControl(final Composite parent) {
+        final Composite composite = new Composite(parent, SWT.NONE);
+        composite.setLayoutData(GridDataFactory.fillDefaults().create());
+        composite.setLayout(GridLayoutFactory.fillDefaults().create());
+        final DataBindingContext dbc = new DataBindingContext();
+        createRadioButtonComposite(composite, dbc);
+        final Composite stackedComposite = new Composite(composite, SWT.NONE);
+        stackLayout = new CustomStackLayout(stackedComposite);
+        stackedComposite.setLayout(stackLayout);
+        stackedComposite.setLayoutData(GridDataFactory.fillDefaults().grab(true, true).create());
+        createbusinessVariableTableViewerComposite(stackedComposite, dbc);
+        createDocumentTableViewerComposite(stackedComposite, dbc);
+        createDocumentNameField(composite, dbc);
+        bindRadioButtonsToComposite(dbc);
+        final MultiValidator multiValidator = new AvailableDataValidator(availableBusinessData, selectedDataObservable, availableDocuments,
+                businessObjectStore);
+        dbc.addValidationStatusProvider(multiValidator);
+        WizardPageSupport.create(this, dbc);
+        setControl(composite);
+    }
+
+    public void createRadioButtonComposite(final Composite parent, final DataBindingContext dbc) {
+        final Composite radioButtonComposite = new Composite(parent, SWT.NONE);
+        radioButtonComposite.setLayoutData(GridDataFactory.fillDefaults().grab(true, false).create());
+        radioButtonComposite.setLayout(GridLayoutFactory.fillDefaults().numColumns(2).margins(20, 20).create());
+        businessVariableButton = new Button(radioButtonComposite, SWT.RADIO);
+        businessVariableButton.setText(Messages.businessVariable);
+        documentButton = new Button(radioButtonComposite, SWT.RADIO);
+        documentButton.setText(Messages.document);
+        selectionTypeObservable = new SelectObservableValue(Boolean.class);
+        selectionTypeObservable.addOption(Boolean.TRUE, SWTObservables.observeSelection(businessVariableButton));
+        selectionTypeObservable.addOption(Boolean.FALSE, SWTObservables.observeSelection(documentButton));
+        if (availableBusinessData.isEmpty()) {
+            businessVariableButton.setEnabled(false);
+        }
+        if (availableDocuments.isEmpty()) {
+            documentButton.setEnabled(false);
+        }
+    }
+
+    public void createbusinessVariableTableViewerComposite(final Composite parent, final DataBindingContext dbc) {
```

Comment:
I'm not sure about disabling add from data when there is no data, because it is not obvious for new users what is done when clicking this button and that he/she has to create data first.